probe-rs/src/debug/variable.rs

```diff
@@ -655,214 +699,239 @@ impl Variable {
+            }
+            _ if first_child.is_none() => {
+                // This is a struct with no children, so just print the type name.
+                // This is for example the None value of an Option or the empty type ().
+                type_name
+            }
+            _ if matches!(
+                self.name,
+                VariableName::StaticScopeRoot
+                    | VariableName::LocalScopeRoot
+                    | VariableName::RegistersRoot
+            ) =>
+            {
+                format_root_value(variable_cache, indentation, children, &type_name)
+            }
+            _ => format_default_value(
+                variable_cache,
+                indentation,
+                &self.name,
+                children,
+                &type_name,
+                show_name,
+            ),
+        })
     }
 
     /// Calculate the memory range that contains the value of this variable.
+    ///
     /// If the location and/or byte size is not known, then return None.
-    /// Note: We don't do any validation of the memory range here, and leave it up to the caller to
-    /// validate the memory ranges before attempting to read them.
+    /// Note: We don't do any validation of the memory range here and leave it
+    /// up to the caller to validate the memory ranges before attempting to read
+    /// them.
     pub fn memory_range(&self) -> Option<Range<u64>> {
-        if let VariableLocation::Address(address) = self.memory_location {
-            if self.byte_size.is_some_and(|byte_size| byte_size.is_zero()) {
-                // This can happen for instance with empty arrays, and even though we don't need to read the
-                // array data, we need to read the array structure type.
-                Some(address..address + 4)
-            } else {
-                self.byte_size
-                    .map(|byte_size| address..(address + byte_size))
-            }
+        let VariableLocation::Address(address) = self.memory_location else {
+            return None;
+        };
+
+        self.byte_size.map(|byte_size| {
+            byte_size
+                .is_zero()
+                .then(|| address..address + 4)
+                .unwrap_or_else(|| address..(address + byte_size))
+        })
+    }
+}
+
+/// Format a pointer value
+///
+/// Formats the pointed to value and potential subsequent children as well.
+fn format_pointer_value(
+    variable_cache: &VariableCache,
+    indentation: usize,
+    first_child: Option<&Variable>,
+) -> String {
+    let line_start = line_indent_string(indentation);
+
+    let value = if let Some(first_child) = first_child {
+        first_child
+            .formatted_variable_value(variable_cache, indentation + 1, true)
+            .expect("a child. This is a bug. Please report it.")
+    } else {
+        "Unable to resolve referenced variable value".to_string()
+    };
+
+    format!("{line_start}{value}",)
```

Comment:
```suggestion
    format!("{line_start}{value}")
```

Reply:
Yeah, rustfmt does not erradicate these and I overlooked this one :/